editor.planx.uk/src/@planx/components/shared/Preview/SummaryList.tsx

```diff
@@ -76,56 +76,119 @@ const components: {
   [TYPES.Statement]: Question,
   [TYPES.TaskList]: undefined,
   [TYPES.TextInput]: TextInput,
-};
+} as const;
 
-interface SummaryListsBySectionsProps extends SummaryListProps {
+type BreadcrumbEntry = [Store.nodeId, Store.breadcrumbs];
+
+interface SummaryListBaseProps {
+  flow: Store.flow;
+  passport: Store.passport;
+  changeAnswer: (id: Store.nodeId) => void;
+  showChangeButton: boolean;
+}
+
+interface SummaryListsBySectionsProps extends SummaryListBaseProps {
+  breadcrumbs: Store.breadcrumbs;
   sectionComponent: React.ElementType<any> | undefined;
 }
 
+interface SummaryBreadcrumb {
+  component: React.FC<ComponentProps>;
+  nodeId: Store.nodeId;
+  userData: Store.userData;
+  node: Store.node;
+}
+
+interface SummaryListProps extends SummaryListBaseProps {
+  summaryBreadcrumbs: SummaryBreadcrumb[];
+}
+
 function SummaryListsBySections(props: SummaryListsBySectionsProps) {
   const [hasSections, getSortedBreadcrumbsBySection] = useStore((state) => [
     state.hasSections,
     state.getSortedBreadcrumbsBySection,
   ]);
 
-  const sections = getSortedBreadcrumbsBySection();
+  const isValidComponent = ([nodeId, userData]: BreadcrumbEntry) => {
+    const node = props.flow[nodeId];
+    const Component = node.type && components[node.type];
 
-  return hasSections ? (
-    <>
-      {sections.map((sectionBreadcrumbs, i) => (
-        <React.Fragment key={i}>
-          <Box sx={{ display: "flex", justifyContent: "space-between" }}>
-            <Typography component={props.sectionComponent || "h2"} variant="h5">
-              {props.flow[`${Object.keys(sectionBreadcrumbs)[0]}`]?.data?.title}
-            </Typography>
-          </Box>
-          <SummaryList
-            breadcrumbs={sectionBreadcrumbs}
-            flow={props.flow}
-            passport={props.passport}
-            changeAnswer={props.changeAnswer}
-            showChangeButton={props.showChangeButton}
-          />
-        </React.Fragment>
-      ))}
-    </>
-  ) : (
-    <SummaryList
-      breadcrumbs={props.breadcrumbs}
-      flow={props.flow}
-      passport={props.passport}
-      changeAnswer={props.changeAnswer}
-      showChangeButton={props.showChangeButton}
-    />
-  );
-}
+    const isPresentationalComponent = Boolean(Component);
```

Comment:
nit: This was helpful for understanding what `components` represents. Maybe a sign that `components` should actually be called `presentationalComponents`?

Reply:
Good shout - updated 👌